File: Data Structures & Algorithms/two-integer-sum/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> um;
        for(int i  = 0; i <nums.size(); ++i){
            int complement = target - nums[i];
            if(um.find(complement) != um.end()){
                return {um[complement] , i};
            }
            um[nums[i]] =  i;
        }
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:

        prefix_sum = 0
        count = 0
        mp = {0: 1}

        for x in nums:
            prefix_sum += x

            if prefix_sum - k in mp:
                count += mp[prefix_sum - k]

            mp[prefix_sum] = mp.get(prefix_sum, 0) + 1

        return count
